### PR TITLE
Fix issue when an empty ArraySegment is a member of a class

### PR DESCRIPTION
**Pull request by ITaluone**
This finishes #2445 

Al the credit goes to @ShaharPrishMSFT

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [x] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [x] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [x] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

**Comment by ITaluone** [comment]
Was only the qodana fix :)

File: Tests/FluentAssertions.Equivalency.Specs/BasicSpecs.cs [hunk 1]
```diff
@@ -645,4 +645,20 @@ public void When_the_graph_contains_guids_it_should_properly_format_them()
         // Assert
         act.Should().Throw<XunitException>().WithMessage("Expected property actual[0].Id*to be *-*, but found *-*");
     }
+
+    [Fact]
+    public void When_a_class_contains_an_empty_array_segment()
```
Comment (jnyrup) [inline Tests/FluentAssertions.Equivalency.Specs/BasicSpecs.cs:650]: Could use a more descriptive name. E.g. `Empty_array_segments_can_be_compared_for_equivalency`

File: docs/_pages/releases.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ sidebar:
 * Improved the failure message for `[Not]HaveExplicitProperty` when wrapped in an `AssertionScope` and not implementing the interface - [#2403](https://github.com/fluentassertions/fluentassertions/pull/2403)
 * Improved the failure message for `[Not]HaveExplicitMethod` when wrapped in an `AssertionScope` and not implementing the interface - [#2403](https://github.com/fluentassertions/fluentassertions/pull/2403)
 * Changed `BeEquivalentTo` to exclude `private protected` members from the comparison - [#2417](https://github.com/fluentassertions/fluentassertions/pull/2417)
-
+* When formatting messages, ignoring ArraySegment<> values (they crash the message generation issue) - [#2445](https://github.com/fluentassertions/fluentassertions/pull/2445), [#2511](https://github.com/fluentassertions/fluentassertions/pull/2511)
```
Comment (jnyrup) [inline docs/_pages/releases.md:36]: I don't know what this sentence means?
This should be on functional level. 
E.g. "Fixed comparing empty `ArraySegment`s for equivalency" or "Fixed using `BeEquivalentTo` on an empty `ArraySegment`".

Comment (ITaluone) [reply docs/_pages/releases.md]: I just cherry-picked all commits, and changed nothing :D

**Review by jnyrup — APPROVED** [review]

**Pull request merged**